File: leetcode-04/solutions/python3/0969.py
class Solution:
    def pancakeSort(self, A: List[int]) -> List[int]:
        n = len(A)

        ans = []
        target = n

        for i in range(n):
            index = A.index(target) + 1
            A[:index] = A[:index][::-1]
            A[:target] = A[:target][::-1]
            ans.append(index)
            ans.append(target)
            target -= 1

        return ans

File: leetcode-04/solutions/cpp/0969.cpp
class Solution {
 public:
  vector<int> pancakeSort(vector<int>& A) {
    const int n = A.size();

    vector<int> ans;
    int target = n;

    for (int i = 0; i < n; i++) {
      int index = find(A, target);
      reverse(A.begin(), A.begin() + index);
      reverse(A.begin(), A.begin() + target);
      ans.push_back(index);
      ans.push_back(target--);
    }

    return ans;
  }

 private:
  int find(vector<int>& A, int target) {
    for (int i = 0; i < A.size(); i++)
      if (A[i] == target) return i + 1;
    return -1;
  }
};